tools/Performance/DynamoPerformanceTests/ResultsComparer.cs

```diff
@@ -0,0 +1,373 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.VisualBasic.FileIO;
+
+namespace DynamoPerformanceTests
+{
+    /// <summary>
+    /// Class that compares two sets of performance results: 
+    /// the "base" results, and the "diff" results.
+    /// </summary>
+    public class ResultsComparer
+    {
+        private class BenchmarkResult
+        {
+            /// <summary>
+            /// Name of the benchmark method
+            /// </summary>
+            internal string Method { get; set; }
+
+            /// <summary>
+            /// Name of test graph
+            /// </summary>
+            internal string Graph { get; set; }
+
+            /// <summary>
+            /// Mean benchmark time
+            /// </summary>
+            internal double Mean { get; set; }
+
+            /// <summary>
+            /// Error in results
+            /// </summary>
+            internal double Error { get; set; }
+
+            /// <summary>
+            /// Standard deviation of results
+            /// </summary>
+            internal double StdDev { get; set; }
+        }
+
+        private class BenchmarkComparison
+        {
+            /// <summary>
+            /// Results of base benchmark
+            /// </summary>
+            internal BenchmarkResult Base;
+
+            /// <summary>
+            /// Results of diff benchmark
+            /// </summary>
+            internal BenchmarkResult Diff;
```

Comment:
The name `Diff` for the current benchmark sounds misleading. It is the new benchmark and not the diff between the baseline, is it?

Reply:
Yeah, I agree with you on this. I've updated the language and replaced `Diff` with `New`.